docs/Features.md

```diff
@@ -0,0 +1,216 @@
+of starting at the unnamed code block, it will start at the code block with this name.
+
+Note that if you use a named entrypoint, there is no way to reference the unnamed code blocks as
+macros. You can, however, use the unnamed code blocks to provide examples, for example, to the
+readers of the documentation, so they are still useful.
+
+## Multiple Files, Multiple Entrypoints
+
+By naming code blocks with prefix `file:` followed by a relative path, multiple code files can be created
+from one source file. Each code block with the `file:` prefix is treated as a separate entry point.
+
+~~~md
+```rust - file:src/lib.rs
+fn say_hello() {
+    println!("Hello Literate Programmer!");
+}
+```
+~~~
+
+## File Transclusions
+
+Yarner supports file transclusion to allow for more structured or modular project sources.
+*This feature is currently only supported for Markdown.*
+
+Here, the content of file `src/main.rs.md` would be pulled into the document
+before compilation:
+
+```md
+@{{[src/main.rs](src/main.rs.md)}}
+```
+which would render as a link in the sources file. Or simply:
+```md
+@{{src/main.rs.md}}
+```
+which would not render as a link.
+
+Unnamed entrypoints are renamed to file entrypoints during transclusion.
+It is not required that transcluded files have their own entrypoints.
+
+All code blocks from transcluded files are accessible from the *transcluding* file,
+as well as from other transcluded files.
+
+## Include linked files
+
+Files linked from the main source document are included in the compilation process.
+*This feature is currently only supported for Markdown.*
+
+As an example, Yarner would also compile the file `src/main.rs.md` here:
+
+```md
+* [src/main.rs](src/main.rs.md)
+```
+
+However, files are only included in the compilation if they are referenced by a *relative path*, and the file exists in that location.
+
+## Multiple languages
+
+Some documentation formats allow you to indicate the language that a code block is written in. In
+fact, it is recommended that you always include the language when you write a code block,
+particularly if multiple programming languages are used within the same document.
+
+By properly labelling all code blocks, it is then possible to write a program in multiple
+programming languages at once. Whether this is practical or not remains to be seen, but it is
+supported nonetheless. By then supplying a language name on the command line, only code blocks in
+that language are used when generating the tangled source. For example, here is a trivial program
+written in two languages:
+
+~~~tex
+Here we have hello world in Ruby:
+
+```rb
+puts "Hello world"
+```
+
+And here it is again in Rust:
+
+```rust
+fn main() {
+  println!("Hello world");
+}
+```
+~~~
+
+Compiling this with no language supplied with just ignore language information, so a single output
+will be generated containing both languages. However, supplying the `--language rb` flag to Yarner
```

Comment:
Is this possible via the TOML as well as it seems like something that will not change for a given project.

Reply:
This is very rarely used, I think.

Reply:
As written above, if this can be considered fully redundant to named entry points, I would probably remove the feature entirely to simplify things.

Reply:
(I also do not rarity applies here: If I do use the feature, then I probably use it each time I compile the documents of a given project. It does not change between invocations.)